File: src/main/java/me/rasztabiga/fridgy/products/service/mapper/RecipeMapper.java
package me.rasztabiga.fridgy.products.service.mapper;

import me.rasztabiga.fridgy.products.domain.*;
import me.rasztabiga.fridgy.products.service.dto.RecipeDTO;
import org.mapstruct.*;

/**
 * Mapper for the entity {@link Recipe} and its DTO {@link RecipeDTO}.
 */
@Mapper(componentModel = "spring", uses = {})
public interface RecipeMapper extends EntityMapper<RecipeDTO, Recipe> {
    @Named("id")
    @BeanMapping(ignoreByDefault = true)
    @Mapping(target = "id", source = "id")
    RecipeDTO toDtoId(Recipe recipe);
}

File: src/main/kotlin/me/rasztabiga/fridgy/products/service/mapper/RecipeMapper.kt
package me.rasztabiga.fridgy.products.service.mapper

import me.rasztabiga.fridgy.products.domain.Recipe
import me.rasztabiga.fridgy.products.service.dto.RecipeDTO
import org.mapstruct.*

/**
 * Mapper for the entity [Recipe] and its DTO [RecipeDTO].
 */
@Mapper(componentModel = "spring", uses = [])
interface RecipeMapper :
    EntityMapper<RecipeDTO, Recipe> {

    @Named("id")
    @BeanMapping(ignoreByDefault = true)
    @Mappings(
        Mapping(target = "id", source = "id")
    )
    fun toDtoId(recipe: Recipe): RecipeDTO
}
